extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +147,181 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!Window} win The window element
+   * @param {!string} frameUrl  The URL of the cross-domain iframe
+   */
+  processCrossDomainIframe(win, frameUrl) {
+    let frameData;
+    if (Transport.hasCrossDomainIframe(this.type_)) {
+      frameData = Transport.getFrameData(this.type_);
+      ++(frameData.usageCount);
+    } else {
+      frameData = this.createCrossDomainIframe(win, frameUrl);
+      win.document.body.appendChild(frameData.frame);
+    }
+    dev().assert(frameData, 'Trying to use non-existent frame');
+  }
+
+  /**
+   * Create a cross-domain iframe for third-party vendor anaytlics
+   * @param {!Window} win The window element
+   * @param {!string} frameUrl  The URL of the cross-domain iframe
+   * @return {!FrameData}
+   * @VisibleForTesting
+   */
+  createCrossDomainIframe(win, frameUrl) {
+    // Explanation of IDs:
+    // Each instance of Transport (owned by a specific amp-analytics tag, in
+    // turn owned by a specific creative) has an ID in this._id.
+    // Each cross-domain iframe also has an ID, stored here in sentinel.
+    // These two types of IDs are drawn from the same pool of numbers, and
+    // are thus mutually unique.
+    // There is a many-to-one relationship, in that several creatives may
+    // utilize the same analytics vendor, so perhaps creatives #1 & #2 might
+    // both use xframe #3.
+    // Of course, a given creative may use multiple analytics vendors, but
+    // in that case it would use multiple amp-analytics tags, so the
+    // transport.id_ -> sentinel relationship is *not* many-to-many.
+    const sentinel = Transport.createUniqueId_();
+    const useLocal = getMode().localDev || getMode().test;
+    const useRtvVersion = !useLocal;
+    const scriptSrc = calculateEntryPointScriptUrl(
+        win.parent.location, 'ampanalytics-lib', useLocal, useRtvVersion);
+    const frameName = JSON.stringify(/** @type {JsonObject} */ ({
+      scriptSrc,
+      sentinel,
+    }));
+    const frame = createElementWithAttributes(win.document, 'iframe',
+        /** @type {!JsonObject} */ ({
+          sandbox: 'allow-scripts',
+          name: frameName,
+          'data-amp-3p-sentinel': sentinel,
+        }));
+    frame.sentinel = sentinel;
+    setStyles(frame, {
+      display: 'none',
+      position: 'absolute',
```

Comment:
Spoke to Keith...if we were creating an amp-iframe, we would want to be sure that it is in the viewport. But since we are creating a plain iframe, this is likely not necessary. Will remove and test to make sure we are correct.